src/nvim/mark_extended.h

```diff
@@ -0,0 +1,65 @@
+#ifndef NVIM_EXTMARK_H
+#define NVIM_EXTMARK_H
+
+#include "nvim/mark_extended_defs.h"
+#include "nvim/lib/kbtree.h"
+#include "nvim/lib/kvec.h"
+#include "nvim/map.h"
+
+typedef PMap(cstr_t) StringMap2; // TODO do these have to be seperate?
+
+typedef struct ExtendedMark ExtendedMark;
+struct ExtendedMark {
+  StringMap2 *names; // key = namespace, value = id
```

Comment:
the map is used for holding all of the names that a mark might have.
```
mark_set('fancyplugin', 'mark1' ..)
mark_set('morefancyplugin', 1 ..)
```
the two calls above will create only one mark (less memory, less stuff to update)
where `mark->names = {fancyplugin: mark1, morefancyplugin: 1}`
On querying all the marks in a range, we just have to  check if the namespace exists for that mark.
afaik maps are the fastest for doing this check?